source/particle/property/elastic_stress.cc

```diff
@@ -0,0 +1,183 @@
+
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Property
+    {
+      template <int dim>
+      ElasticStress<dim>::ElasticStress ()
+      {}
+
+      template <int dim>
+      void
+      ElasticStress<dim>::initialize ()
+      {
+        AssertThrow((Plugins::plugin_type_matches<const MaterialModel::ViscoPlastic<dim>>(this->get_material_model())
+                     ||
+                     Plugins::plugin_type_matches<const MaterialModel::Viscoelastic<dim>>(this->get_material_model())),
+                    ExcMessage("This particle property only makes sense in combination with the viscoelastic or visco_plastic material model."));
+
+        AssertThrow(this->get_parameters().enable_elasticity == true,
+                    ExcMessage ("This particle property should only be used if 'Enable elasticity' is set to true"));
+
+      }
+
+      template <int dim>
+      void
+      ElasticStress<dim>::initialize_one_particle_property(const Point<dim> &position,
+                                                           std::vector<double> &data) const
+      {
+        // Give each elastic stress field its initial composition if one is prescribed.
+        data.push_back(this->get_initial_composition_manager().initial_composition(position,this->introspection().compositional_index_for_name("stress_xx")));
+
+        data.push_back(this->get_initial_composition_manager().initial_composition(position,this->introspection().compositional_index_for_name("stress_yy")));
+
+        if (dim == 2)
+          {
+            data.push_back(this->get_initial_composition_manager().initial_composition(position,this->introspection().compositional_index_for_name("stress_xy")));
+          }
+        else if (dim == 3)
+          {
+            data.push_back(this->get_initial_composition_manager().initial_composition(position,this->introspection().compositional_index_for_name("stress_zz")));
+
+            data.push_back(this->get_initial_composition_manager().initial_composition(position,this->introspection().compositional_index_for_name("stress_xy")));
+
+            data.push_back(this->get_initial_composition_manager().initial_composition(position,this->introspection().compositional_index_for_name("stress_xz")));
+
+            data.push_back(this->get_initial_composition_manager().initial_composition(position,this->introspection().compositional_index_for_name("stress_yz")));
+          }
+
+      }
+
+      template <int dim>
+      void
+      ElasticStress<dim>::update_particle_property(const unsigned int data_position,
+                                                   const Vector<double> &solution,
+                                                   const std::vector<Tensor<1,dim> > &gradients,
+                                                   typename ParticleHandler<dim>::particle_iterator &particle) const
+      {
+        MaterialModel::MaterialModelInputs<dim> in(1, this->n_compositional_fields());
```

Comment:
You could make this a mutable class member like I did in Property::ViscoPlasticStrainInvariant. This would save some time allocating these vectors for every particle. Same for the MaterialModelOutputs object. But I guess that since you need to call the whole expensive evaluate function it will not matter much, this property is going to be expensive either way.

Reply:
I gave this a shot and we'll see if I did it correctly. However, in the process I did have to add an `override` to `update_particle_property` in `include/aspect/particle/property/composition.h` to suppress a compiler warning. I'm not sure exactly why that came up now, but will revert if this was not the correct fix or I implemented something incorrectly on my end.